### DISCUSSION TITLE
no distribution folder after running mvn clean package

### DISCUSSION THREAD

**Discussion by earebano01** [Pi4J/pi4j#302] [Q&A]
Hi All,

I'm trying to do the [MinimalExample](https://github.com/Pi4J/pi4j-example-minimal/tree/main) and have been struggling trying to make it work for few days now, at the moment, i was able to go as far as running mvn clean install, but the there's no distribution folder on target folder that contains the run.sh like on the guide which i have read over and over, any help is highly appreciated, here's maven and java version, pom.xml, raspberry.properties and the execution :

pi@raspberrypi:~/projet1/projet1/target$ mvn -v
Apache Maven 3.6.3
Maven home: /usr/share/maven
Java version: 11.0.20.1, vendor: Ubuntu, runtime: /usr/lib/jvm/java-11-openjdk-arm64
Default locale: en, platform encoding: UTF-8
OS name: "linux", version: "5.15.0-1038-raspi", arch: "aarch64", family: "unix"

pi@raspberrypi:~/projet1/projet1/target$ java -version
openjdk version "11.0.20.1" 2023-08-24
OpenJDK Runtime Environment (build 11.0.20.1+1-post-Ubuntu-0ubuntu122.04)
OpenJDK 64-Bit Server VM (build 11.0.20.1+1-post-Ubuntu-0ubuntu122.04, mixed mode)

-------------------------
pom.xml
-------------------------

<?xml version="1.0" encoding="UTF-8"?>
<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
    <modelVersion>4.0.0</modelVersion>
    <groupId>com.projet1</groupId>
    <artifactId>projet1</artifactId>
    <version>1.0-SNAPSHOT</version>
    <packaging>jar</packaging>
    <name>Raspi Project</name>

    
    
    <properties>
        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
        <maven.compiler.source>11</maven.compiler.source>
        <maven.compiler.target>11</maven.compiler.target>
        <target.platform.name>raspberry</target.platform.name>
        <run.with.java.module>true</run.with.java.module>
        
        <jvmargs.run></jvmargs.run>
        <jvmargs.debug></jvmargs.debug>

        <slf4j.version>1.7.32</slf4j.version>
        <pi4j.version>2.1.0</pi4j.version>
    </properties>

    <dependencies>
        <dependency>
            <groupId>org.slf4j</groupId>
            <artifactId>slf4j-api</artifactId>
            <version>${slf4j.version}</version>
        </dependency>
        <dependency>
            <groupId>org.slf4j</groupId>
            <artifactId>slf4j-simple</artifactId>
            <version>${slf4j.version}</version>
        </dependency>

        
        <dependency>
            <groupId>com.pi4j</groupId>
            <artifactId>pi4j-core</artifactId>
            <version>${pi4j.version}</version>
        </dependency>

        
        <dependency>
            <groupId>com.pi4j</groupId>
            <artifactId>pi4j-plugin-raspberrypi</artifactId>
            <version>${pi4j.version}</version>
        </dependency>
        <dependency>
            <groupId>com.pi4j</groupId>
            <artifactId>pi4j-plugin-pigpio</artifactId>
            <version>${pi4j.version}</version>
        </dependency>
    </dependencies>
        
    <build>
        <finalName>${project.artifactId}</finalName>
        <plugins>
            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-compiler-plugin</artifactId>
                <version>3.8.1</version>
                <configuration>
                    <release>${maven.compiler.source}</release>
                    <target>${maven.compiler.target}</target>
                    <showDeprecation>true</showDeprecation>
                    <showWarnings>true</showWarnings>
                    <verbose>false</verbose>
                </configuration>
            </plugin>
            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-dependency-plugin</artifactId>
                <version>3.1.2</version>
                <executions>
                    <execution>
                        <id>copy-dependencies</id>
                        <phase>prepare-package</phase>
                        <goals>
                            <goal>copy-dependencies</goal>
                        </goals>
                        <configuration>
                            <stripVersion>true</stripVersion>
                            <outputDirectory>${project.build.directory}/lib</outputDirectory>
                        </configuration>
                    </execution>
                </executions>
            </plugin>
            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-jar-plugin</artifactId>
                <version>3.2.0</version>
                <configuration>
                    <archive>
                        <manifest>
                            <addClasspath>true</addClasspath>
                            <classpathLayoutType>custom</classpathLayoutType>
                            <customClasspathLayout>lib/${artifact.artifactId}.${artifact.extension}</customClasspathLayout>
                            <mainClass>com.projet1.Main</mainClass>
                        </manifest>
                    </archive>
                    <outputDirectory>${finalJarDir}</outputDirectory>
                </configuration>
            </plugin>
            
            <plugin>
                <artifactId>maven-antrun-plugin</artifactId>
                <version>3.0.0</version>
                <executions>
                    <execution>
                        <id>setMavenJarPluginOutputDir</id>
                        <phase>compile</phase>
                        <configuration>
                            <exportAntProperties>true</exportAntProperties>
                            <target>
                                <condition property="finalJarDir" value="${project.build.directory}/lib" else="${project.build.directory}">
                                    <equals arg1="${run.with.java.module}" arg2="true" casesensitive="" />
                                </condition>
                            </target>
                        </configuration>
                        <goals>
                            <goal>run</goal>
                        </goals>
                    </execution>
                    <execution>
                        <id>clean</id>
                        <phase>clean</phase>
                        <configuration>
                            <target>
                                <property name="local.dist.dir" value="${project.build.directory}" />        
                                <property name="remote.runtime.encoding" value="UTF-8" />
                                <property name="dist.jar.name" value="${project.build.finalName}" />        
                                <ant antfile="antrun/build.xml" target="clean-remote" />
                            </target>
                        </configuration>
                        <goals>
                            <goal>run</goal>
                        </goals>
                    </execution>
                    <execution>
                        <id>build</id>
                        <phase>install</phase>
                        <configuration>
                            <target>
                                <property name="local.dist.dir" value="${project.build.directory}" />        
                                <property name="remote.runtime.encoding" value="UTF-8" />
                                <property name="dist.jar.name" value="${project.build.finalName}" />        
                                <ant antfile="antrun/build.xml" target="build-remote" />
                            </target>
                        </configuration>
                        <goals>
                            <goal>run</goal>
                        </goals>
                    </execution>
                    <execution>
                        <id>exec</id>
                        <configuration>
                            <target>
                                <property name="remote.run.jvmargs" value="${jvmargs.run}" />
                                <property name="target.run.as.module" value="${run.with.java.module}" />
                                <property name="local.dist.dir" value="${project.build.directory}" />        
                                <property name="remote.runtime.encoding" value="UTF-8" />
                                <property name="dist.jar.name" value="${project.build.finalName}" />        
                                <ant antfile="antrun/build.xml" target="run-remote" />
                            </target>
                        </configuration>
                        <goals>
                            <goal>run</goal>
                        </goals>
                    </execution>
                    <execution>
                        <id>debug</id>
                        <configuration>
                            <target>
                                <property name="remote.debug.jvmargs" value="${jvmargs.debug}" />
                                <property name="target.run.as.module" value="${run.with.java.module}" />
                                <property name="local.dist.dir" value="${project.build.directory}" />        
                                <property name="remote.runtime.encoding" value="UTF-8" />
                                <property name="dist.jar.name" value="${project.build.finalName}" />        
                                <ant antfile="antrun/build.xml" target="debug-remote" />
                            </target>
                        </configuration>
                        <goals>
                            <goal>run</goal>
                        </goals>
                    </execution>
                </executions>
                <dependencies>
                    <dependency>
                        <groupId>com.jcraft</groupId>
                        <artifactId>jsch</artifactId>
                        <version>0.1.55</version>
                    </dependency>
                    <dependency>
                        <groupId>org.apache.ant</groupId>
                        <artifactId>ant-jsch</artifactId>
                        <version>1.10.8</version>
                    </dependency>
                </dependencies>
            </plugin>
        </plugins>
    </build>
</project>

-----------------------
raspberry.properties
-----------------------

#
# Please configure this to describe the remote RPI machine
#
# Connection data
target.platform.host=192.168.5.173
target.platform.port=22
target.platform.username=USERNAME
# if the password is set, it is used for connecting via SSH to RPI
# to authenticate with your certificate, comment the password and configure the privatekey and passphrase properties
target.platform.password=PASSWORD
#target.platform.privatekey=<private key file>
#target.platform.passphrase=<private key passphrase>
# remote folders
# the location of the RPI JRE
target.remote.jre=/usr/lib/jvm/java-11-openjdk-arm64/bin/java
target.debug.supported=true
target.debug.port=1955
target.run.as.root=true
# The remote target root folder where your projects is deployed
target.remote.home=/home/pi/projet1/projet1/target
target.run.module.class=com.projet1.projet1/com.projet1.Main

-------------------------------------
execution of mvn clean package
-------------------------------------

[INFO] Scanning for projects...
[INFO] 
[INFO] ------------------------< com.projet1:projet1 >-------------------------
[INFO] Building Raspi Project 1.0-SNAPSHOT
[INFO]   from pom.xml
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- clean:3.2.0:clean (default-clean) @ projet1 ---
[INFO] Deleting C:\Users\USERNAME\Documents\folder\folder\folder\Projets\projet1\projet1\target
[INFO]
[INFO] --- antrun:3.0.0:run (clean) @ projet1 ---
[INFO] Executing tasks
[INFO]   [sshexec] Connecting to 192.168.5.173:22
[INFO]   [sshexec] cmd : cd '/home/pi/projet1/projet1/target/projet1/dist'; rm -rf *.jar lib/*.jar
bash: line 1: cd: /home/pi/projet1/projet1/target/projet1/dist: No such file or directory
[INFO] Executed tasks
[INFO]
[INFO] --- resources:3.3.1:resources (default-resources) @ projet1 ---
[INFO] Copying 1 resource from src\main\resources to target\classes
[INFO]
[INFO] --- compiler:3.8.1:compile (default-compile) @ projet1 ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 3 source files to C:\Users\USERNAME\Documents\folder\folder\folder\Projets\projet1\projet1\target\classes
[WARNING] /C:/Users/USERNAME/Documents/folder/folder/folder/Projets/projet1/projet1/src/main/java/module-info.java:[1,19] module name component projet1 should avoid terminal digits
[WARNING] /C:/Users/USERNAME/Documents/folder/folder/folder/Projets/projet1/projet1/src/main/java/module-info.java:[1,11] module name component projet1 should avoid terminal digits
[INFO]
[INFO] --- antrun:3.0.0:run (setMavenJarPluginOutputDir) @ projet1 ---
[INFO] Executing tasks
[INFO] Executed tasks
[INFO]
[INFO] --- resources:3.3.1:testResources (default-testResources) @ projet1 ---
[INFO] skip non existing resourceDirectory C:\Users\USERNAME\Documents\folder\folder\folder\Projets\projet1\projet1\src\test\resources
[INFO]
[INFO] --- compiler:3.8.1:testCompile (default-testCompile) @ projet1 ---
[INFO] No sources to compile
[INFO]
[INFO] --- surefire:3.1.2:test (default-test) @ projet1 ---
[INFO] No tests to run.
[INFO]
[INFO] --- dependency:3.1.2:copy-dependencies (copy-dependencies) @ projet1 ---
[INFO] Copying slf4j-api-1.7.32.jar to C:\Users\USERNAME\Documents\folder\folder\folder\Projets\projet1\projet1\target\lib\slf4j-api.jar
[INFO] Copying slf4j-simple-1.7.32.jar to C:\Users\USERNAME\Documents\folder\folder\folder\Projets\projet1\projet1\target\lib\slf4j-simple.jar
[INFO] Copying pi4j-core-2.1.0.jar to C:\Users\USERNAME\Documents\folder\folder\folder\Projets\projet1\projet1\target\lib\pi4j-core.jar
[INFO] Copying pi4j-plugin-raspberrypi-2.1.0.jar to C:\Users\USERNAME\Documents\folder\folder\folder\Projets\projet1\projet1\target\lib\pi4j-plugin-raspberrypi.jar
[INFO] Copying pi4j-plugin-pigpio-2.1.0.jar to C:\Users\USERNAME\Documents\folder\folder\folder\Projets\projet1\projet1\target\lib\pi4j-plugin-pigpio.jar
[INFO] Copying pi4j-library-pigpio-2.1.0.jar to C:\Users\USERNAME\Documents\folder\folder\folder\Projets\projet1\projet1\target\lib\pi4j-library-pigpio.jar
[INFO] 
[INFO] --- jar:3.2.0:jar (default-jar) @ projet1 ---
[INFO] Building jar: C:\Users\USERNAME\Documents\folder\folder\folder\Projets\projet1\projet1\target\lib\projet1.jar
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  4.734 s
[INFO] Finished at: 2023-10-17T16:35:18-03:00

**Comment by FDelporte** [comment]
The command is `mvn clean package`, not `mvn clean install` but when I try both create target/distribution on my side...

```
pi4j-example-minimal % ls -l target/distribution 
total 1184
-rw-r--r--  1 frankdelporte  staff  301816 Oct 17 21:53 pi4j-core-2.3.0.jar
-rw-r--r--  1 frankdelporte  staff    6827 Oct 17 21:53 pi4j-example-minimal-0.0.1.jar
-rw-r--r--  1 frankdelporte  staff  155886 Oct 17 21:53 pi4j-library-pigpio-2.3.0.jar
-rw-r--r--  1 frankdelporte  staff   37487 Oct 17 21:53 pi4j-plugin-pigpio-2.3.0.jar
-rw-r--r--  1 frankdelporte  staff   26009 Oct 17 21:53 pi4j-plugin-raspberrypi-2.3.0.jar
-rwxr-xr-x  1 frankdelporte  staff     101 Oct 17 21:53 run.sh
-rw-r--r--  1 frankdelporte  staff   41513 Oct 17 21:53 slf4j-api-1.7.32.jar
-rw-r--r--  1 frankdelporte  staff   15257 Oct 17 21:53 slf4j-simple-1.7.32.jar
```

**Reply by earebano01** [reply]
Hi Frank,

Yes sorry, that's the correct command, sorry for the typo, i did it again on my raspberry pi and the result is the same for me, do you have any other suggestion on where i should look for some configurations that i might have overlooked? I actually tried to do the same project using all the instructions on the site for learning purposes and i uploaded it on my git and cloning the file from (https://github.com/earebano01/projet1) to my rpi and here's the results :

pi@raspberrypi:~/projet1/projet1$ mvn clean package
[INFO] Scanning for projects...
[INFO]
[INFO] ------------------------< com.projet1:projet1 >-------------------------
[INFO] Building Raspi Project 1.0-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO]
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ projet1 ---
[INFO] Deleting /home/pi/projet1/projet1/target
[INFO]
[INFO] --- maven-antrun-plugin:3.0.0:run (clean) @ projet1 ---
[INFO] Executing tasks
[INFO]   [sshexec] Connecting to 192.168.5.173:22
[INFO]   [sshexec] cmd : cd '/home/pi/projet1/projet1/target/projet1/dist'; rm -rf *.jar lib/*.jar
bash: line 1: cd: /home/pi/projet1/projet1/target/projet1/dist: No such file or directory
[INFO] Executed tasks
[INFO]
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ projet1 ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 1 resource
[INFO]
[INFO] --- maven-compiler-plugin:3.8.1:compile (default-compile) @ projet1 ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 3 source files to /home/pi/projet1/projet1/target/classes
[WARNING] /home/pi/projet1/projet1/src/main/java/module-info.java:[1,19] module name component projet1 should avoid terminal digits
[WARNING] /home/pi/projet1/projet1/src/main/java/module-info.java:[1,11] module name component projet1 should avoid terminal digits
[INFO]
[INFO] --- maven-antrun-plugin:3.0.0:run (setMavenJarPluginOutputDir) @ projet1 ---
[INFO] Executing tasks
[INFO] Executed tasks
[INFO]
[INFO] --- maven-resources-plugin:2.6:testResources (default-testResources) @ projet1 ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory /home/pi/projet1/projet1/src/test/resources
[INFO]
[INFO] --- maven-compiler-plugin:3.8.1:testCompile (default-testCompile) @ projet1 ---
[INFO] No sources to compile
[INFO]
[INFO] --- maven-surefire-plugin:2.12.4:test (default-test) @ projet1 ---
[INFO] No tests to run.
[INFO]
[INFO] --- maven-dependency-plugin:3.1.2:copy-dependencies (copy-dependencies) @ projet1 ---
[INFO] Copying slf4j-api-1.7.32.jar to /home/pi/projet1/projet1/target/lib/slf4j-api.jar
[INFO] Copying slf4j-simple-1.7.32.jar to /home/pi/projet1/projet1/target/lib/slf4j-simple.jar
[INFO] Copying pi4j-core-2.1.0.jar to /home/pi/projet1/projet1/target/lib/pi4j-core.jar
[INFO] Copying pi4j-plugin-raspberrypi-2.1.0.jar to /home/pi/projet1/projet1/target/lib/pi4j-plugin-raspberrypi.jar
[INFO] Copying pi4j-plugin-pigpio-2.1.0.jar to /home/pi/projet1/projet1/target/lib/pi4j-plugin-pigpio.jar
[INFO] Copying pi4j-library-pigpio-2.1.0.jar to /home/pi/projet1/projet1/target/lib/pi4j-library-pigpio.jar
[INFO]
[INFO] --- maven-jar-plugin:3.2.0:jar (default-jar) @ projet1 ---
[INFO] Building jar: /home/pi/projet1/projet1/target/lib/projet1.jar
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  16.217 s
[INFO] Finished at: 2023-10-17T20:12:40Z
[INFO] ------------------------------------------------------------------------
pi@raspberrypi:~/projet1/projet1$ ls -l target/
total 24
drwxrwxr-x 2 pi pi 4096 Oct 17 20:12 antrun
drwxrwxr-x 3 pi pi 4096 Oct 17 20:12 classes
drwxrwxr-x 3 pi pi 4096 Oct 17 20:12 generated-sources
drwxrwxr-x 2 pi pi 4096 Oct 17 20:12 lib
drwxrwxr-x 2 pi pi 4096 Oct 17 20:12 maven-archiver
drwxrwxr-x 3 pi pi 4096 Oct 17 20:12 maven-status
pi@raspberrypi:~/projet1/projet1$

**Comment by FDelporte** [comment]
at first sight, your pom files doesn't have the distribution steps that are in the minimal example, eg.
https://github.com/Pi4J/pi4j-example-minimal/blob/main/pom.xml#L117
